includes/rest-api.php

```diff
@@ -1,6 +1,54 @@
 <?php
 namespace tenup;
 
+/**
+ * Check the REST request before it's dispatched to see if it is approved for access
+ * @param WP_Error|null|bool $result The original REST result
+ * @param WP_REST_Server $rest_server Instance of the REST server
+ * @param WP_REST_Request $request The current REST request
+ * @return $result The original null request if successful or a string denying the request if it is not an approved route
+ */
+function rest_pre_dispatch($result, $rest_server, $request) {
+	$approved_routes = apply_filters( 'tenup_rest_approved_routes', $rest_server->get_routes() );
+
+	foreach( $approved_routes as $route => $data ) {
+		if( preg_match( '@^' . $route . '$@i', $request->get_route(), $matches ) ) {
+			return $result;
+		}
+	}
+
+	$result = 'your request is denied.';
```

Comment:
Worth adding a more specific error message here to help anyone who isn't aware of this feature debug it? Perhaps something like `The 10up Experience plugin has disabled access to this endpoint.`

Reply:
I agree with @pcrumm about the contents of the message. I'm also wondering what the rest of the response looks like when using this filter - does it still 403 or is it a 200 or something else?